pkg/thumbnail/thumbnail.go

```diff
@@ -84,3 +84,42 @@ func BuildStaticThumbnail(inputBuf []byte, resp *http.Response) ([]byte, error)
 
 	return outputBuf, nil
 }
+
+func BuildAnimatedThumbnail(inputBuf []byte, resp *http.Response) ([]byte, error) {
+	image, err := vips.NewImageFromBuffer(inputBuf)
+
+	if err != nil {
+		return []byte{}, fmt.Errorf("could not load image from url: %s", resp.Request.URL)
+	}
+
+	maxThumbnailSize := int(cfg.MaxThumbnailSize)
+	format := image.Format()
+
+	if image.Width() <= maxThumbnailSize && image.Height() <= maxThumbnailSize {
+		return inputBuf, nil
+	}
+
+	importParams := vips.NewImportParams()
+
+	// n=-1 is used for animated images to make sure to get all frames and not just the first one.
+	if format == vips.ImageTypeGIF || format == vips.ImageTypeWEBP {
+		importParams.NumPages.Set(-1)
+	}
+
+	image, err = vips.LoadThumbnailFromBuffer(inputBuf, maxThumbnailSize, maxThumbnailSize, vips.InterestingAll, vips.SizeDown, importParams)
+
+	if err != nil {
+		fmt.Println(err)
+		return []byte{}, fmt.Errorf("could not transform image from url: %s", resp.Request.URL)
+	}
+
+	// We export to WebP by default to save on bandwidth and cache storage.
+	exportParams := vips.NewWebpExportParams()
+	outputBuf, _, err := image.ExportWebp(exportParams)
+
+	if err != nil {
+		return []byte{}, fmt.Errorf("could not export image from url: %s", resp.Request.URL)
+	}
```

Comment:
Should use a logger derived with logger.FromContext() here. (Requires passing down the request context to BuildAnimatedThumbnail() but this is considered good practice anyway.)